### PR TITLE
Feature/atr 852 dev forbid async apis

### PR DESCRIPTION
**Pull request by SENya1990**
Changes Overview:
- added new banned async APIs to PX1099 diagnostic
- fixed lookup of banned APIs for extension methods
- fixed lookup of banned APIs for generic methods and generic types
- updated diagnostic reporting to pass user friendly symbol representation to the diagnostic 
- added unit tests for banned async APIs usage
- added unit tests for API parsing from string
- updated documentation

### CODE REVIEW

File: docs/diagnostics/PX1099.md [hunk 1]
```diff
@@ -86,22 +86,30 @@ embedded in the Acuminator assemblies. To disable the PX1099 diagnostic, you nee
...
-| `System.IO.File`                                   |   Type    |      ISV     | Usage of the File type is forbidden in Acumatica customization projects.                  |
-| `System.GC`                                        |   Type    |      ISV     | Access to garbage collector is forbidden in Acumatica customization projects. |
-| `System.OperatingSystem`                           |   Type    |      ISV     | Usage of the OperatingSystem type is forbidden in Acumatica customization projects.       |
-| `System.Reflection.MethodInfo`                     |   Type    |      ISV     | Usage of reflection is forbidden in Acumatica customization projects.            |
-| `PX.Common.WebConfig`                              |   Type    |      ISV     | Access to the WebConfig type is forbidden in Acumatica customization projects.    |
+|                             API Name                        | API Type  | API Ban Type |                           Ban Reason                                                                                                         |
```
Comment (SENya1990) [inline docs/diagnostics/PX1099.md:89]: @EAndrosova please review ban reasons for new banned APIs

File: src/Acuminator/Acuminator.Utilities/BannedApi/Data/BannedApis.txt [hunk 2]
```diff
@@ -13,4 +13,25 @@ T:PX.Common-WebConfig ISV Access to the WebConfig type is forbidden in Acumatica
...
 M:System-Math.Round(System.Double,System.Int32) General Math.Round uses Banker's Rounding by default, which rounds to the closest even number. Usually, this is not the desired rounding behavior. Use the Math.Round overload with the MidpointRounding parameter to explicitly specify the desired rounding behavior.
-M:System-Math.Round(System.Decimal,System.Int32) General Math.Round uses Banker's Rounding by default, which rounds to the closest even number. Usually, this is not the desired rounding behavior. Use the Math.Round overload with the MidpointRounding parameter to explicitly specify the desired rounding behavior.
\ No newline at end of file
+M:System-Math.Round(System.Decimal,System.Int32) General Math.Round uses Banker's Rounding by default, which rounds to the closest even number. Usually, this is not the desired rounding behavior. Use the Math.Round overload with the MidpointRounding parameter to explicitly specify the desired rounding behavior.
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Utilities/BannedApi/Data/BannedApis.txt:16]: @EAndrosova please review ban reasons for new banned APIs

**Review by EAndrosova — APPROVED** [review]

File: src/Acuminator/Acuminator.Utilities/BannedApi/Data/BannedApis.txt (hunk 2 above)
Comment (EAndrosova) [reply src/Acuminator/Acuminator.Utilities/BannedApi/Data/BannedApis.txt:16]: Edited in a separated commit

**Comment by SENya1990** [comment]
Do not merge until actual ban reasons are edited

**Comment by SENya1990** [comment]
Updated actual ban reasons and tests

**Review by dropsonic — APPROVED** [review]

**Pull request merged**